### PR TITLE
Wilson/distributor voucher data

### PR DESCRIPTION
**Pull request by wilsonj806**
## Functionality Added Or Changed
- Updated fetch all gift cards route to return more data
- Updated rspec test for fetch all distributor gift cards
- Added a route to fetch voucher metadata for a distributor
- Migrate Campaigns table so distributor_id cannot be null

## Fetch gift cards response structure
Note: this doesn't include QR code URL yet
```json
{
  "gift_cards": [
     {
        "id": 1,
        "expiration": "whenever",
        "address1": "whenever",
        "address2": "whenever",
        "city": "whenever",
        "distibutor_id": "whenever",
        "seller_gift_card_id": "whenever",
        "state": "whenever",
        "value": "whenever",
        "zip_code": "whenever",
     }
  ],
  "seller_names": {
    [seller_id]: { "id": 1, "en": "Shunfa-bakery", "zh-cn": "zhCN name here" },
    //... etc
  }
}
```

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -212,7 +212,7 @@
 end
 
 seller = Seller.find_by(seller_id: 'shunfa-bakery')
-contact = Contact.find_or_create_by!(name: 'Apex for Youth', email: 'distributor@apexforyouth.com')
+contact = Contact.find_or_create_by!(name: 'Apex for Youth', email: ENV['SEED_DIST_EMAIL'])
```
Comment (jacoblb64) [inline db/seeds.rb:215]: can you add this to the .env.example so folks know they have to fill it out to seed?

File: db/migrate/20210405210045_make_campaign_distributor_id_non_null.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class MakeCampaignDistributorIdNonNull < ActiveRecord::Migration[6.0]
```
Comment (jacoblb64) [inline db/migrate/20210405210045_make_campaign_distributor_id_non_null.rb:1]: maybe also run `rubocop -a` on these new files

**Review by jacoblb64 — APPROVED** [review]
Beefy queries! Pull latest `master` to resolve your build issue -- @stanzheng had to update some broken dependencies. Thanks for taking care of this!

File: db/seeds.rb (hunk 1 above)
Comment (wilsonj806) [reply db/seeds.rb:215]: Added in 8b2fc85

File: db/migrate/20210405210045_make_campaign_distributor_id_non_null.rb (hunk 2 above)
Comment (wilsonj806) [reply db/migrate/20210405210045_make_campaign_distributor_id_non_null.rb:3]: Went a little overboard and ran rubocop on all the files sooooo oops. But resolved

**Comment by jacoblb64** [comment]
[[Dist. Tools] Build out the Tabular View](https://trello.com/c/IcahXScR/808-dist-tools-build-out-the-tabular-view)

File: .env.example [hunk 3]
```diff
@@ -14,6 +14,10 @@ THINK_CHINATOWN_APPLICATION_ID=<>
 THINK_CHINATOWN_ACCESS_TOKEN=<>
 THINK_CHINATOWN_WEBHOOK_SIGNATURE_KEY=<>
 
+# SEED_DIST_EMAIL is for local development
+# and for attaching your email to the seeded Distributor
+SEED_DIST_EMAIL=
```
Comment (jacoblb64) [inline .env.example:19]: you might have to add this to the .github workflow if this needs to be defined for tests

**Review by jacoblb64 — APPROVED** [review]
Looks good to me! I know all the formatting changes are scary, but they're good to have in here.

One other thing to note is I think you need to pull latest master as there were some breaking changes to the CI tests you definitely need to get your pipeline to pass

File: .env.example (hunk 3 above)
Comment (wilsonj806) [reply .env.example:19]: Fixed, also re failing tests: they're from the webhooks/ charges tests since I'm merging from my forked branch.

**Pull request merged**